xarray/core/missing.py

```diff
@@ -203,6 +203,7 @@ def __init__(
 
         self.method = method
         self.cons_kwargs = kwargs
+        del self.cons_kwargs["axis"]
```

Comment:
why is "axis" being sent to this level if it isn't needed? This suggests we need to delete it elsewhere